zml/ops.zig

```diff
@@ -638,3 +639,49 @@ pub fn identityCustomCall(name: [:0]const u8, input: Tensor, context: *anyopaque
     );
     return Tensor._result(input.shape(), op.result(0));
 }
+
+/// At runtime the given tensor will be materialized and copied to host,
+/// and the callback will be called on it.
+pub fn addHostCallback(
+    callback: *const fn (HostBuffer) void,
+    input: Tensor,
+) Tensor {
+    // TODO: implement addCallback that exposes a pjrt.Buffer, so that the user can decide if they need to copy.
+    if (input.getContext().target() != .cuda) return input;
+
+    const len = input.byteSize();
+    // Reserve memory to be able to log the runtime Buffer later during the computation.
+    // This memory is leaked, we currently have no way to tie this lifetime to the lifetime of the module being compiled.
```

Comment:
Just a note about that, this is something solved with the `api_version` 2 that we will switch to eventually

Reply:
yes, can't wait for you to add support for that :-)